### PR TITLE
Remove gray background for transparent images

### LINKED ISSUE #4316: Transparent images have grey background
## Bug Description

After #3935, transparent PNGs have an unexpected grey background in the  editor.

## Expected Behaviour

Transparent pngs should not have a transparent background

## Steps to Reproduce

1. Insert transparent image on canvas

## Screenshots

Screenshot 2020-09-02 at 20 45 18

## Additional Context

- Plugin Version: `main`
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by merapi**
## Summary

Removes placeholder (bg color for now) for fully loaded images (we can also do that or cached/small images, but we can improve that once we have a clear vision for the placeholder/loading bar).

## Relevant Technical Choices

Small `resourceList` refactor to prevent transparent images placeholder flickering in some cases.

I've added a new resource with a transparent background to the media gallery and corresponding Percy test for it.

## Testing Instructions

PNGs with transparent background should be displayed correctly (when fully loaded).

Not like this:
Screenshot 2020-09-02 at 20 45 18

---

Fixes #4316

### CODE REVIEW

File: assets/src/edit-story/utils/resourceList.js [hunk 1]
```diff
@@ -19,6 +19,18 @@
...
+  get: function (id) {
+    return this.list[id];
+  },
+  set: function (id, value) {
+    if (value?.type === 'cached' && this.list[id]) {
+      // We already have better (or equal) resource than cached, prevent flickering
+      return;
+    }
+    this.list[id] = value;
+  },
```
Comment (barklund) [inline assets/src/edit-story/utils/resourceList.js]: Could we unit test this? It'll be super simple, but just to document the desired behavior twice.

**Review by barklund — APPROVED** [review]
Seems to work excellently. There might be a very short flash of large gray square, but it's barely noticeable.

**Comment by merapi** [comment]
> Seems to work excellently. There might be a very short flash of large gray square, but it's barely noticeable.

Yep, that part can be improved _"once we have a clear vision for the placeholder/loading bar"_.
This PR is mostly about fixing the full-size image.

**Comment by csossi** [comment]
Verified in QA

**Pull request merged**